src/config.rs

```diff
@@ -1174,14 +1174,23 @@ impl Default for RaftDefaultCfConfig {
 }
 
 impl RaftDefaultCfConfig {
-    pub fn build_opt(&self, cache: &Option<Cache>) -> ColumnFamilyOptions {
+    pub fn build_opt(&self, cache: &Option<Cache>, gc_on_compaction: bool) -> ColumnFamilyOptions {
         let no_region_info_accessor: Option<&RegionInfoAccessor> = None;
         let mut cf_opts = build_cf_opt!(self, CF_DEFAULT, cache, no_region_info_accessor);
         let f = Box::new(FixedPrefixSliceTransform::new(region_raft_prefix_len()));
         cf_opts
             .set_memtable_insert_hint_prefix_extractor("RaftPrefixSliceTransform", f)
             .unwrap();
         cf_opts.set_titandb_options(&self.titan.build_opts());
+        if gc_on_compaction {
```

Comment:
I prefer to always install the factory. And then examine the `GC_IN_COMPACTION_FILTER` in `create_compaction_filter`. Then this feature can be enabled/disabled dynamically.

Reply:
That's a good idea, but some extra work needed. It's safe to dynamically enable/disable the config (not break consistency), but dynamically disabling it may leave some log entries without tombstone and not deleted. An extra machism needed to handle it.

Reply:
If the region is compacted again, those log will be deleted. Seems it's OK?

Reply:
Enabling `gc-on-compaction` will stop inserting tombstone when deleting. If the previous config enabled `gc-on-comaction` and then disable it before all log entries between `from` and `to` is deleted by compaction filter, these log entries will not be able to be deleted again (raftstore will track index `to`), expect enable `gc-on-compaction` again.